### ISSUE TITLE
Question - Support for newer mongo DB version? (4.x is EOL)

### ISSUE THREAD

**Issue by jamess60** [Yooooomi/your_spotify#223]
Hi 

I see on the read me "Some ARM-based devices might have trouble with Mongo >= 5. I suggest you use the image mongo:4.4.", assuming I am running this on an x86 platform, is there any reason not to use a newer version?

The Mongo version in the compose example (4.4.8) is almost 3 years old (2021-08), id like to run something newer if possible.

**Comment by jamess60** [comment]
Note - it may be worth adding the findings of this to the readme

**Comment by Yooooomi** [comment]
Hello, I think you can use any newer version as the driver I use is always up to date. I think you can easily try 5.x.
Let me know what happens

**Comment by CrowMash** [comment]
I'm running YourSpotify on mongo:5 and it looks just fine (upgraded on 01/03/23).
I tried mongo:6 but I had error and not have the time to look into at the time.

**Comment by jamess60** [comment]
Hi both

Sorry for the slow reply, busy week. 

I tried going from 4.4.8 (latest 4.x) to 5.0.16 (latest 5.x) on amd64/Linux. Not only did this work fine, but I saw significant improvements in performance when using the last year or all views (I have over a decade of history with average of 90k mins listened per year so its a big db...) 

Loading last year went from almost 1 minute to under 10 seconds. 

I tried 6.0.5 but the login page gave the red banner saying unable to communicate with server. The your spotify client and your spotify mongo containers both achieved running status, but the your spotify server container got stuck on starting. 

I reverted to :5 (the latest 5.x version), same tag @CrowMash is using, zero issues. 

While I am not able to easily validate this on ARM hardware, I think it would be wise to update the readme to :5 as this gives a huge performance uplift and is not EOL. 

Are there any plans to introduce support for mongo 6?

**Comment by Yooooomi** [comment]
Hello guys, thanks a lot for the benchmark and the tests. I will definitely look forward to implement mongo 6. If I remember correctly, mongo 4 & 5 were not compatible with arm devices at that time. Maybe better luck with 6 with the arm hype we're currently living.

**Comment by Yooooomi** [comment]
You have to have the Spotify app and the login process. The nightly image supports mongo 6. 
To migrate to mongo 6 start the stack using mongo 5, the server will migrate the database. Then stop it and use the mongo 6 image.
Be sure to backup your database before doing this.

**Comment by lorenzopicoli** [comment]
I'm running mongo 6 with no problems after upgrading like @Yooooomi described

**Comment by ikaruswill** [comment]
Hi folks, just wanted to chyme in that MongoDB 5.0+ requires AVX Instruction set extension. 

```
WARNING: MongoDB 5.0+ requires a CPU with AVX support, and your current system does not appear to have that!
  see https://jira.mongodb.org/browse/SERVER-54407
  see also https://www.mongodb.com/community/forums/t/mongodb-5-0-cpu-intel-g4650-compatibility/116610/2
  see also https://github.com/docker-library/mongo/issues/485#issuecomment-891991814
```

I'm running `your_spotify` on low-power CPUs ([J5005](https://www.intel.sg/content/www/xa/en/products/sku/128984/intel-pentium-silver-j5005-processor-4m-cache-up-to-2-80-ghz/specifications.html) and [N5105](https://www.intel.sg/content/www/xa/en/products/sku/212328/intel-celeron-processor-n5105-4m-cache-up-to-2-90-ghz/specifications.html)) and AVX is not supported. 

Only the latest Alder Lake N series processor supports it ([N100](https://www.intel.sg/content/www/xa/en/products/sku/231803/intel-processor-n100-6m-cache-up-to-3-40-ghz/specifications.html)).

Thought I'd raise it up here for awareness for all others who are running low-power x86 in their homelabs.

**Comment by Yooooomi** [comment]
Thanks, i added a note in the release notes

**Comment by jamess60** [comment]
Hi all

Sorry for the slow update. 

@ikaruswill Very good find. Im running mine on E5-2697V2s so doubt this is a blocker in my case, but could certainly see that being problematic for some. 

I am still running the :latest tags of the client and server, given they were last committed to git 2 months ago, I assume they are indeed latest despite not re-pulling for a few days.

 Looking at my reply on Apr 15th, I went from 4 latest to 5 latest to any version of 6 and it didn't work, but going back to 5 was fine. It looks like I accidentally did the process outlined by @Yooooomi  - I guess this is therefore still something that only works on nightly? If so, any idea when this will make it to stable?

**Comment by Yooooomi** [comment]
Hello. This should be stable by now. Could you try repulling?

**Comment by jamess60** [comment]
Hi @Yooooomi 

I updated the image and tested again. I was already on mongo 5, so I stepped to 6.0 -> 6.0.6 -> latest and it appears to work fine!  Thanks for fixing this :)  - Will close the issue 


TLDR of the thread for newcomers: 

- Previous image only supported mongo 4 and limited success with early builds of 5
- Yoooomi pushed an update to enable support for mongo 5/6 
- If you start out on 4, go to 5.0, latest 5, 6.0, latest 6 you will have no issues (unsure if brand new instances can go direct to 6, not tested)
- Mongo 5/6 seem far more performant than 4 when working with large DBs
- Mongo 6 requires the AVX instruction set, so may not play nice with older/lower power x86 chips (atoms etc) or ARM chips.
